### DISCUSSION TITLE
Behaviour of object expiration on files with prefixes

### DISCUSSION THREAD

**Discussion by flyattimes** [minio/minio#15515] [Q&A] [answered]
I have question on the basic functioning on object expiration when file prefixes are used.
I believe I am missing out on something obvious here, any tip is appreciated.

We have the following setup:
Bucket name: bucket1
Test file name: file.zip
Test file prefix: foo/bar/baz/

If I list the objects in bucket1, using recursive = false, this suggests that each part of the prefix is also an object.
For example this:
`for file in self.client.list_objects("bucket1", include_user_meta=True, recursive=False):
print(file.object_name)`

will yield something like:
`foo/`

Now, we would like to use the object expiration feature for all files in bucket1.
We want to avoid a scenario in which `foo/` is treated as an object that has the object expiration rule applied and all of its underlying contents (in this case, file.zip) is removed. 
Instead, we expect only the individual file `file.zip` to be expired and removed. 

Setting the prefix in the lifecycle's configuration is not the best options, since we have too many combinations of `foo/` `bar/` and `baz/`

**Comment by klauspost** [comment, marked as answer]
It is not an object, it is a prefix. It only exists since `foo/bar/baz/file.zip` exists.

When the object is removed the prefix also disappears. As long as there is any object in `foo/...` it will keep existing.

**Reply by flyattimes** [reply]
Wonderful, thanks for clarifying.
